torch/distributed/_sharded_tensor/api.py

```diff
@@ -0,0 +1,206 @@
+from typing import List
+
+import torch
+import torch.distributed as dist
+from torch.distributed._sharding_spec import (
+    ChunkShardingSpec,
+    ShardMetadata,
+    ShardingSpec,
+)
+from torch.distributed._sharding_spec._internals import is_valid_device
+from torch.distributed.utils import _parse_remote_device
+
+
+class Shard(object):
+    """
+    Container which holds the data for a shard as a Tensor and also
+    the associated metadata for that shard.
+    """
+    __slots__ = ['tensor', 'metadata']
```

Comment:
It seems that we cannot add `__slots__` here:

ValueError: 'tensor' in __slots__ conflicts with class variable